app/src/main/kotlin/com/wire/android/notification/NotificationDismissReceiver.kt

```diff
@@ -0,0 +1,48 @@
+package com.wire.android.notification
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.id.toConversationId
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class NotificationDismissReceiver : BroadcastReceiver() {
+
+    @Inject
+    @KaliumCoreLogic
+    lateinit var coreLogic: CoreLogic
+
+    @Inject
+    lateinit var dispatcherProvider: DispatcherProvider
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val conversationId: String? = intent.getStringExtra(EXTRA_CONVERSATION_ID)
+
+        GlobalScope.launch(dispatcherProvider.io()) {
+            val currentSession = coreLogic.getAuthenticationScope().session.currentSession.invoke()
```

Comment:
Using the current session here is incorrect, since the app may receive notifications for multiple authenticated accounts.
You need a way to get the session for the account that the notification is directed to.